django/db/models/sql/compiler.py

```diff
@@ -16,9 +16,15 @@
 from django.db.transaction import TransactionManagementError
 from django.utils.functional import cached_property
 from django.utils.hashable import make_hashable
+from django.utils.regex_helper import _lazy_re_compile
 
 
 class SQLCompiler:
+    ordering_parts = _lazy_re_compile(
+        r'^(.*)\s(ASC|DESC)(.*)',
```

Comment:
In my searching, only group 1 is used, so perhaps can avoid unnecessary capturing:

```suggestion
        r'^(.*)\s(?:ASC|DESC).*',
```

Here are the results of the search:

```
django/db/models/sql/compiler.py
381              without_ordering = self.ordering_parts.search(sql).group(1)
394                  without_ordering = self.ordering_parts.search(sql).group(1)
```

Reply:
Well spotted. And very slightly faster

```
In [7]: capturing = re.compile(r'^(.*)\s(ASC|DESC)(.*)')

In [8]: %timeit capturing.search("foo bar ASC baz")
477 ns ± 18.8 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [9]: non_capturing = re.compile(r'^(.*)\s(?:ASC|DESC).*')

In [10]: %timeit non_capturing.search("foo bar ASC baz")
422 ns ± 1.06 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```